### DISCUSSION TITLE
primary key on domain_metadata table

### DISCUSSION THREAD

**Discussion by raunakkumar** [uber/cadence#4798] [General]
hi, 
We are planning to use cadence as sql/postgres database for our implementation and require to define a primary key for all the tables. We noticed domain_metadata is missing a primary key. Can `notification_version` be a primary key for the table?
```
cadence=# \d domain_metadata;
                 Table "public.domain_metadata"
        Column        |  Type  | Collation | Nullable | Default 
----------------------+--------+-----------+----------+---------
 notification_version | bigint |           | not null | 

```

**Comment by longquanzheng** [comment]
Yes that would work. However usually ppl would expect a immutable pk. You can add a new column as PK, eg id.

**Reply by raunakkumar** [reply]
Thanks @longquanzheng. That was my next suggestion. Hope that wouldn't break anything for cadence.

**Reply by longquanzheng** [reply]
No it won’t. You can also submit a PR to change for every one. I will review it for you
